### ISSUE TITLE
Websocket disconnects (Binance issue, but maybe we can defend against)

### ISSUE THREAD

**Issue by kevflynn** [jaggedsoft/node-binance-api#86]
When subscribing to all ticker trades, I've noticed sometimes some of the tickers become disconnected without a warning or any error message and never reconnect. To discover this issue, but I put a watch dog that listens to trades for each ticker and when the ticker hasn't received a trade in over a minute, I log it to the console. From there, I picked out a few tickers that hadn't received a trade in over 5 minutes and compared to their actual Binance activity and sure enough, they had active trades over the course of the last 5 minutes, ensuring some kind of disconnect on the receiving end. 

That being said - any ideas for better detecting this disconnect so we can keep trying to reconnect until we get a response? Or at least be better aware of what we're actually connected to?

**Comment by jaggedsoft** [comment]
Which websocket functions are having this problem? I am aware of issues with `trades` and `candlesticks` and will be fixing those shortly

**Comment by jaggedsoft** [comment]
It appears what happened last night is binance disconnected everybody, and was down for a short time. Thus, the one and only reconnection attempt failed.
I believe we need a system to continue attempting reconnections periodically until successful

**Comment by kevflynn** [comment]
Got it, thanks - I’m seeing it with trades. I saw the other re-opened issue so if you want to close this one I can just follow along there? Also happy to help if you’re not already working on it.

**Comment by jaggedsoft** [comment]
Thanks friend :) I am busy for now but I may have time later today or tomorrow
The trades and candlesticks issues should be easy to fix so i can see about hammering that out in a few hours

**Comment by jaggedsoft** [comment]
Just to clarify, the trades function that's not reconnecting is this one?
Just fixed a reconnection issue with `candlesticks` function
Will make auto-reconnect more robust as I have more time

#### Get Trade Updates via WebSocket
```javascript
binance.websockets.trades(['BNBBTC', 'ETHBTC'], (trades) => {
	let {e:eventType, E:eventTime, s:symbol, p:price, q:quantity, m:maker, a:tradeId} = trades;
	console.log(symbol+" trade update. price: "+price+", quantity: "+quantity+", maker: "+maker);
});
```

**Comment by kevflynn** [comment]
Right - that's the one. I haven't tried the candlesticks one, but that one is not reconnecting nor was it giving any indication of a disconnect.

**Comment by jaggedsoft** [comment]
Thank you for the report
This bug was resolved in `v0.4.7`
Problem: after the second reconnect it would stop trying to reconnect anymore

Test code:
```js
binance.websockets.trades(['BNBBTC', 'ETHBTC'], (trades) => {
	let {e:eventType, E:eventTime, s:symbol, p:price, q:quantity, m:maker, a:tradeId} = trades;
	console.log(symbol+" trade update. price: "+price+", quantity: "+quantity+", maker: "+maker);
});
setInterval(function() {
	// Terminate all websocket endpoints, every 6 sec
	let endpoints = binance.websockets.subscriptions();
	for ( let endpoint in endpoints ) {
		console.log("..terminating websocket: "+endpoint);
		let ws = endpoints[endpoint];
		ws.terminate();
	}
}, 6000);
```

**Comment by kevflynn** [comment]
👏
